commit ad1a0a4f02ed7882a42580c51ecd8153a353ca37
Author: David Howells <dhowells@redhat.com>
Date:   Tue May 23 21:54:06 2017 -0400

    FS-Cache: Initialise stores_lock in netfs cookie
    
    
    [ Upstream commit 62deb8187d116581c88c69a2dd9b5c16588545d4 ]
    
    Initialise the stores_lock in fscache netfs cookies.  Technically, it
    shouldn't be necessary, since the netfs cookie is an index and stores no
    data, but initialising it anyway adds insignificant overhead.
    
    Signed-off-by: David Howells <dhowells@redhat.com>
    Reviewed-by: Jeff Layton <jlayton@redhat.com>
    Acked-by: Steve Dickson <steved@redhat.com>
    Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
    Signed-off-by: Sasha Levin <alexander.levin@verizon.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/fs/fscache/netfs.c b/fs/fscache/netfs.c
index 9b28649..a8aa00b 100644
--- a/fs/fscache/netfs.c
+++ b/fs/fscache/netfs.c
@@ -48,6 +48,7 @@ int __fscache_register_netfs(struct fscache_netfs *netfs)
 	cookie->flags		= 1 << FSCACHE_COOKIE_ENABLED;
 
 	spin_lock_init(&cookie->lock);
+	spin_lock_init(&cookie->stores_lock);
 	INIT_HLIST_HEAD(&cookie->backing_objects);
 
 	/* check the netfs type is not already present */
